### PR TITLE
Unify three bluetooth de-init tasks and allow exit on shutdown request

### PR DESCRIPTION
**Pull request by laurensvalk**
We have previously implemented the MicroPython VM abort to exit MicroPython under all circumstances, but we have recently added a few pbio tasks that were started after this, and they are not cancellable. This prevents shutdown.

This unifies awaiting on the deinitialization of the bluetooth peripheral, bluetooth broadcasting/observing and bluetooth stdout.

It also adds a break for a shutdown request so the hub can shut down even if a pbio task is stuck (e.g. https://github.com/pybricks/support/issues/1419).

This also finishes implementing the remote disconnect task, which on some platforms was only started but not awaited inside the task. This could be used for a dedicated awaitable remote disconnect method in a future release, which has been requested several times.

This is largely untested, which is a work in progress.

### CODE REVIEW

File: lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c [hunk 1]
```diff
@@ -1051,17 +1061,16 @@ static PT_THREAD(peripheral_disconnect_task(struct pt *pt, pbio_task_t *task)) {
     if (peri->con_handle != NO_CONNECTION) {
         PT_WAIT_WHILE(pt, write_xfer_size);
         GAP_TerminateLinkReq(peri->con_handle, 0x13);
-        PT_WAIT_UNTIL(pt, hci_command_status);
+        PT_WAIT_UNTIL(pt, peri->con_handle == NO_CONNECTION);
```
Comment (dlech) [inline lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c:1064]: I would keep the waiting for command status. The disconnect could happen for a different reason other than the request made here, so the program flow would continue while the command is still being processed on the BT chip.

File: pybricks/pybricks.c [hunk 2]
```diff
@@ -133,10 +136,25 @@ void pb_package_pybricks_init(bool import_all) {
...
-    pb_type_BLE_cleanup();
+    pb_type_ble_start_cleanup();
     #endif
     // Disconnect from remote.
     #if PYBRICKS_PY_PUPDEVICES
-    pb_type_Remote_cleanup();
+    pb_type_lwp3device_start_cleanup();
     #endif // PYBRICKS_PY_PUPDEVICES
+
+    #if PYBRICKS_PY_COMMON_BLE && PYBRICKS_PY_PUPDEVICES
```
Comment (dlech) [inline pybricks/pybricks.c:146]: Should this be `||` instead of `&&`?

File: lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c (hunk 1 above)
Comment (laurensvalk) [reply lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c:1064]: So (in nominal conditions) are we guaranteed to get two events and always in that order?

File: pybricks/pybricks.c (hunk 2 above)
Comment (laurensvalk) [reply pybricks/pybricks.c:146]: Thanks. I'll probably just have to be specific and add `PYBRICKS_PY_PUPDEVICES_REMOTE` to account for the virtualhub which has pupdevices but no remote, and then we can use `||`.

File: lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c (hunk 1 above)
Comment (dlech) [reply lib/pbio/drv/bluetooth/bluetooth_stm32_cc2640.c:1064]: yes. and even if the disconnect event happens first due to non-nominal conditions `peri->con_handle == NO_CONNECTION` is still going to be true after the status event.

**Pull request merged**